esx_service/utils/auth_data.py

```diff
@@ -0,0 +1,306 @@
+
+import sqlite3
+import uuid
+import os
+
+class DbConnectionError(Exception):
+    """ An exception thrown when connection to a sqlite database fails """
+
+    def __init__(self, db_path):
+        self.db_path = db_path
+
+    def __str__(self):
+        return "DB connection error %s" % self.db_path
+
+class DockerVolumeTenant:
+
+    def __init__(self, name, description, default_datastore, default_privileges,
+                    vms, privileges, id=None):
+            self.name = name
+            self.description = description
+            self.default_datastore = default_datastore
+            self.default_privileges = default_privileges
+            self.vms = vms
+            self.privileges = privileges
+            if not id:
+                self.id = str(uuid.uuid4())
+            else:
+                self.id = id
+        
+    def add_vm_to_tenant(self, conn, vm_id, vm_name):
+        tenant_id = self.id
+        conn.execute(
+                "insert into vms(vm_id, vm_name, tenant_id) values (?, ?, ?)",
+                (vm_id, vm_name, tenant_id)
+        )
+        conn.commit()
+
+    def remove_vm_from_tenant(self, conn, vm_id, vm_name):
+        tenant_id = self.id
+        conn.execute(
+                "delete from vms where vm_id=? AND tenant_id=?", 
+                (vm_id, tenant_id)
+        )
+        conn.commit()
+
+    def set_name(self, conn, name):
+        tenant_id = self.id
+        conn.execute(
+                "update tenants set name=? where id=?", 
+                (name, tenant_id)
+        )
+        conn.commit()
+    
+    def set_description(self, conn, description):
+        tenant_id = self.id
+        conn.execute(
+                "update tenants set description=? where id=?", 
+                (description, tenant_id)
+            )
+        conn.commit()
+        
+    def set_default_datastore_and_privileges(self, conn, datastore, privileges):
+        tenant_id = self.id
+        exist_default_datastore = self.default_datastore
+        conn.execute(
+                "update tenants set default_datastore=? where id=?", 
+                (datastore, tenant_id)
+        )
+
+        # remove the old entry
+        conn.execute(
+                "delete from privileges where tenant_id=? and datastore=?", 
+                [tenant_id, exist_default_datastore]
+        )
+
+        for p in privileges:
+            p['tenant_id'] = tenant_id
+        conn.executemany(
+            """
+            insert into privileges values
+            (:tenant_id, :datastore, :global_visibility, :create_volume,
+            :delete_volume, :mount_volume, :max_volume_size, :usage_quota)
+            """,
+            privileges
+        )
+
+        conn.commit()
+            
+    def set_datastore_access_privileges(self, conn, privileges):
+        # If we want to change privileges for <tenant1, ds1> and <tenant2, ds2>
+        # Do we call this function one time or two times?
```

Comment:
Currently, it is an array of dicts. So it would be possible to call once to set privileges for multiple <tenant, datastore> pairs. I think this is more flexible.